http/method.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+/**
+ * Contains the constant {@linkcode HTTP_METHODS} and the type
+ * {@linkcode HttpMethods} and the type guard {@linkcode isHttpMethod} for
+ * working with HTTP methods with type safety.
+ *
+ * @module
+ */
+
+/** A constant array of common HTTP methods. */
+export const HTTP_METHODS = [
+  "HEAD",
+  "OPTIONS",
+  "GET",
+  "PUT",
+  "PATCH",
+  "POST",
+  "DELETE",
```

Comment:
I wonder if we should also list `CONNECT` and `TRACE` which are mentioned in RFC 9110 https://www.rfc-editor.org/rfc/rfc9110.html#section-9.1

(RFC 9110 also mentions [HTTP Method Registry](https://www.iana.org/assignments/http-methods/http-methods.xhtml), which list 39 methods)

Reply:
(Note: Node.js http.METHODS array has 34 items. 28 items overlap with the IANA list, 6 items don't.)

Reply:
It makes sense to align to Node.js `http` methods.